tests/integration_tests.rs

```diff
@@ -0,0 +1,844 @@
+        .unwrap();
+    let recipient2: AccountId = format!("recipient2.{}", near_sandbox::config::DEFAULT_GENESIS_ACCOUNT)
+        .parse()
+        .unwrap();
+
+    // Buy storage first
+    let num_records = 5;
+    let storage_cost = NearToken::from_yoctonear(11_880_000_000_000_000_000_000);
+
+    near_api::Contract(contract_id.clone())
+        .call_function("buy_storage", json!({ "num_records": num_records }))
+        .unwrap()
+        .transaction()
+        .deposit(storage_cost)
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // Submit payment list
+    let payments = vec![
+        json!({
+            "recipient": recipient1.to_string(),
+            "amount": "1000000000000000000000000" // 1 NEAR
+        }),
+        json!({
+            "recipient": recipient2.to_string(),
+            "amount": "2000000000000000000000000" // 2 NEAR
+        }),
+    ];
+
+    let submit_result = near_api::Contract(contract_id.clone())
+        .call_function(
+            "submit_list",
+            json!({
+                "token_id": "native",
+                "payments": payments
+            }),
+        )
+        .unwrap()
+        .transaction()
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap();
+
+    submit_result.assert_success();
+    // List IDs start from 0
+    let list_id: u64 = 0;
+
+    // Verify storage credits were deducted
+    let credits: NearToken = near_api::Contract(contract_id.clone())
+        .call_function(
+            "view_storage_credits",
+            json!({ "account_id": user_id }),
+        )
+        .unwrap()
+        .read_only()
+        .fetch_from(&network_config)
+        .await
+        .unwrap()
+        .data;
+
+    assert_eq!(
+        credits.as_yoctonear(),
+        (num_records - 2) as u128,
+        "Storage credits should be deducted"
+    );
+
+    // Approve the list with exact deposit
+    let total_amount = NearToken::from_yoctonear(3_000_000_000_000_000_000_000_000); // 3 NEAR
+    near_api::Contract(contract_id.clone())
+        .call_function("approve_list", json!({ "list_ref": list_id }))
+        .unwrap()
+        .transaction()
+        .deposit(total_amount)
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // View the list to verify status
+    let list: serde_json::Value = near_api::Contract(contract_id.clone())
+        .call_function("view_list", json!({ "list_ref": list_id }))
+        .unwrap()
+        .read_only()
+        .fetch_from(&network_config)
+        .await
+        .unwrap()
+        .data;
+
+    assert_eq!(list["status"], "Approved");
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_batch_processing() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
create one similar test for fungible tokens, and one for intents. See https://github.com/NEAR-DevHub/near-treasury/blob/staging/playwright-tests/tests/intents/payment-request-ui.spec.js for an example of deploying the intents contracts and fungible tokens contracts to the sandbox ( it is in JS, but you should be able to convert it to Rust easily). For the fungible token test you can use the wNEAR contract. For the Intents test, use USDC on Base network ( like you can see from the javascript sandbox example ).